python/ray/train/session.py

```diff
@@ -195,6 +207,7 @@ def checkpoint(self, **kwargs):
             kwargs = {}
         else:
             kwargs = self._auto_fill_checkpoint_metrics(kwargs)
+            kwargs = self._encode_checkpoint_fn(kwargs)
 
         result = TrainingResult(TrainingResultType.CHECKPOINT, kwargs)
```

Comment:
Is it intentional that this is only being done for checkpoints and not report? Maybe not as high priority because the users probably shouldn't be reporting the model, but I could imagine they might report some tensors?

Reply:
Done!